lib/zendesk_api/collection.rb

```diff
@@ -188,10 +190,24 @@ def fetch!(reload = false)
       end
       path_query_link = (@query || path)
 
-      @response = get_response(path_query_link)
+      if intentional_obp_request?
+        warn "OBP will be deprecated after Oct 2023."
+      elsif @next_page.nil?
+        @options_per_page_was = @options.delete("per_page")
+        @options.page = { size: (@options_per_page_was || DEFAULT_PAGE_SIZE) }
+      end
+
+      begin
+        @response = get_response(path_query_link)
+      rescue ZendeskAPI::Error::NetworkError => e
+        raise e if intentional_obp_request?
+        @options.per_page = @options_per_page_was
+        @options.page = nil
+        @response = get_response(path_query_link)
+      end
```

Comment:
we're adding more than 10LOC in a method that's already long, we should extract this